### PR TITLE
Update ORTModule feature with remaining PRs from feature branch

### PR DESCRIPTION
**Pull request by thiagocrepaldi**
This is a PR that merges additional changes from ORTModule dev branch into master

### CODE REVIEW

File: orttraining/orttraining/python/training/ortmodule.py [hunk 1]
```diff
@@ -33,6 +34,12 @@ def _ortvalue_to_dlpack(ortvalue):
 
 def _ortvalue_from_dlpack(dlpack_tensor):
     return OrtValue(C.OrtValue.from_dlpack(dlpack_tensor))
+class Verbosity(IntEnum):
```
Comment (mrry) [inline orttraining/orttraining/python/training/ortmodule.py:37]: Nit: Add 2 blank lines before class definition (looks like a merge artefact).

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 2]
```diff
@@ -144,11 +144,11 @@ stages:
 
 
   - ${{ if eq(parameters.enable_linux_gpu, true) }}:
-    - job: Linux_py_GPU_Wheels
-      timeoutInMinutes: 120
+    - job: Linux_py_Wheels_openmp
```
Comment (mrry) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:147]: Is this change correct? It looks like the condition is still based on `enable_linux_gpu`, but the changes below seem to switch to a CPU build.

File: .gitignore [hunk 3]
```diff
@@ -24,16 +24,16 @@ nuget_root/
 *.code-workspace
 __pycache__
 onnxruntime_profile*.json
-/docs/python/*.md
-/docs/python/auto_examples/*
-/docs/python/media/*
-/docs/python/examples/*.onnx
-/docs/python/examples/graph.*
+/docs/python/inference/*.md
```
Comment (mrry) [inline .gitignore:27]: I'm not sure what depends on this, but can we avoid changing the path for inference docs?

(This seems to be related to a lot of file moves down below as well.)

File: docs/python/training/content.rst [hunk 4]
```diff
@@ -0,0 +1,54 @@
+This document describes ORTModule PyTorch frontend API for the ONNX Runtime (aka ORT) training acelerator.
```
Comment (mrry) [inline docs/python/training/content.rst:1]: Will this show up in any of our websites?

Should we add a note to this file to inform readers that the ORTModule is currently under development, and the API may change?

**Review by mrry — DISMISSED** [review]

File: docs/python/training/content.rst (hunk 4 above)
Comment (thiagocrepaldi) [reply docs/python/training/content.rst:1]: Not yet. this first PR only introduces all the tooling required to produce python API doc from a build. This content is a mere placeholder for content to come

Next PRs will actually include all relevant content

File: .gitignore (hunk 3 above)
Comment (thiagocrepaldi) [reply .gitignore:27]: Unfortunately the move can't be avoided to keep a single docs folder.
The way sphinx works, it takes ownership of the entire root folder (docs in this case) folder and its subfolders. creating a docs/training subfolder would result in training doc going into inference folder

Future PRs might unify both docs in a single one, but this first PR only enables inference and training to coexist

Comment (mrry) [reply .gitignore:27]: Can we rollback that change and propose it as a PR against master then? It seems like this is the only inference-affecting part of the PR, and we shouldn't hold up the rest of the merge on it.

File: orttraining/orttraining/python/training/ortmodule.py (hunk 1 above)
Comment (thiagocrepaldi) [reply orttraining/orttraining/python/training/ortmodule.py:37]: Running autopep8 in the whole file

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml (hunk 2 above)
Comment (thiagocrepaldi) [reply tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:147]: Moving doc generation to a separate PR

**Review by baijumeswani — APPROVED** [review]

File: .gitignore (hunk 3 above)
Comment (thiagocrepaldi) [reply .gitignore]: ps: the directory change had been approved by Pranav and Xavier Dupre (author of doc gen on Inference side)

**Review by mrry — APPROVED** [review]

**Pull request merged**